purchase_advance_payment/models/account_move.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def action_post(self):
+        # Automatic reconciliation of payment when bill confirmed.
+        res = super().action_post()
+        if bool(
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("purchase_advance_payment.auto_reconcile_advance_payments")
+        ):
+            purchase_order = self.mapped("line_ids.purchase_order_id")
+            if (
+                purchase_order
+                and self.invoice_outstanding_credits_debits_widget is not False
```

Comment:
`self` can be a recordset here, and in that case `self.invoice_outstanding_credits_debits_widget` will raise a `ValueError`